format/spec.md

```diff
@@ -241,7 +245,14 @@ Struct evolution requires the following rules for default values:
 
 #### Column Projection
 
-Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids. If a field id is missing from a data file, its value for each row should be `null`.
+Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids.
+
+Values for field ids which are not present in a data file must be resolved according the following rules:
+
+* Return the value from partition metadata if an [Identity Transform](#partition-transforms) exists for the field and the partition value is present in the `partition` struct on `data_file` object in the manifest. 
+* Use `schema.name-mapping.default` metadata to map field id to columns without field id as described below and use the column if it is present.
+* Return the default value if it has a defined in `initial-default` (See [Default values](#default-values) section for more details). 
+* Return `null` in all other cases.
```

Comment:
[minor] should we move these lines after we explain what `schema.name-mapping.default` is ?

Reply:
I referenced that it is described below.  I placed it i moved text from the following paragraph and highlights the overall logic is non-trivial.